elyra/pipeline/pipeline_definition.py

```diff
@@ -326,6 +336,7 @@ def __init__(
         if validate:
             self.validate()
 
+        self.coerce_kv_pair_properties()
```

Comment:
Per your earlier comment in our side conversation, I think it would be better to "protect" this method (with an _ prefix) and call it within `propagate_pipeline_default_properties()` since its used exclusively there.  Looking at the two methods, they do appear to overlap wrt to walking the pipeline properties and nodes and I wonder if the KVL coercion could be performed _as_ the properties are being walked thereby eliminating the need for a separate method (and an additional pass across the properties).  Might be worth exploring.

Reply:
Thanks for the reminder - this slipped my mind!